plugins/openai_plugin.py

```diff
@@ -48,6 +62,35 @@ def _parse_args(self, args):
 
         logger.debug("Model name: %s", self.model_name)
 
+        self.api = args.get('api')
+
+        if not self.api:
+            self.api = 'chat' if "/v1/chat/completions" in self.host else 'legacy'
+
+        if self.api not in APIS:
+            logger.error("Invalid api type: %s", self.api)
+
+        # TODO Make this configurable
+        self.request_defaults = dict(
+            temperature = 0.0,
+            seed = 42,
+        )
+
+    def _process_resp(self, resp: bytes) -> Optional[dict]:
+        try:
+            _, found, data = resp.partition(b"data: ")
+            if not found:
+                return None
+            message = json.loads(data)
+            logger.debug("Message: %s", message)
+            return message
+        except json.JSONDecodeError:
+            logger.exception("Response line could not be json decoded: %s", resp)
+        except KeyError:
+            logger.exception(
+                "KeyError, unexpected response format in line: %s", resp
+            )
```

Comment:
I don't think that `KeyError` can be raised in this `try` block 🤔

Reply:
My bad, leftover from the old code. Removed it and the similar one in the non-streaming function.